From: Ben Hutchings <ben@decadent.org.uk>
Subject: [PATCH] [xen][i386] Fix kernel logging via userspace

Xen has its own private copy of update_process_times() for i386.
As part of the fix for #510478 this needs to call printk_tick() too.

--- a/arch/x86/kernel/time_32-xen.c
+++ b/arch/x86/kernel/time_32-xen.c
@@ -585,6 +585,7 @@
 	run_local_timers();
 	if (rcu_pending(cpu))
 		rcu_check_callbacks(cpu, user_mode_vm(get_irq_regs()));
+	printk_tick();
 	scheduler_tick();
 	run_posix_cpu_timers(current);
 	profile_tick(CPU_PROFILING);
